src/EnergyPlus/DataSizing.cc

```diff
@@ -649,88 +653,301 @@ Real64 calcDesignSpecificationOutdoorAir(EnergyPlusData &state,
+            }
         }
     } else {
         // This is a DesignSpecification:OutdoorAir:SpaceList
-        Real64 sumAreaOA = 0.0;
-        Real64 sumArea = 0.0;
         for (int dsoaCount = 1; dsoaCount <= this->numDSOA; ++dsoaCount) {
             auto const &thisDSOA = state.dataSize->OARequirements(this->dsoaIndexes(dsoaCount));
             int const dsoaSpaceNum = this->dsoaSpaceIndexes(dsoaCount);
             if (thisDSOA.OAFlowMethod != OAFlowCalcMethod::PerPerson && thisDSOA.OAFlowMethod != OAFlowCalcMethod::PerZone &&
                 thisDSOA.OAFlowMethod != OAFlowCalcMethod::ACH) {
                 if ((spaceNum == 0) || (spaceNum == dsoaSpaceNum)) {
                     Real64 spaceArea = state.dataHeatBal->space(this->dsoaSpaceIndexes(dsoaCount)).FloorArea;
-                    sumArea + -spaceArea;
-                    sumAreaOA += thisDSOA.OAFlowPerArea * spaceArea;
+                    Real64 spaceOAArea = thisDSOA.OAFlowPerArea * spaceArea;
+                    if (useMinOASchFlag && (thisDSOA.oaFlowFracSched != nullptr)) {
+                        spaceOAArea *= thisDSOA.oaFlowFracSched->getCurrentVal();
+                    }
+                    sumAreaOA += spaceOAArea;
                 }
             }
         }
-        if ((spaceNum == 0) && (state.dataHeatBal->Zone(zoneNum).FloorArea)) {
-            desFlowPA = sumAreaOA / state.dataHeatBal->Zone(zoneNum).FloorArea;
-        } else if (sumArea > 0.0) {
+    }
+    auto &thisZone = state.dataHeatBal->Zone(zoneNum);
+    sumAreaOA = sumAreaOA * thisZone.Multiplier * thisZone.ListMultiplier;
+    return sumAreaOA;
+}
+
+Real64 OARequirementsData::floorArea(EnergyPlusData &state, int const zoneNum,
+                                     int const spaceNum) // Space index (if applicable)
+{
+    Real64 sumArea = 0.0; // Zone or space floor area [m2]
+    if (this->numDSOA == 0) {
+        // This is a simple DesignSpecification:OutdoorAir
+        if (spaceNum == 0) {
+            sumArea = state.dataHeatBal->Zone(zoneNum).FloorArea;
+        } else {
+            sumArea = state.dataHeatBal->space(spaceNum).FloorArea;
+        }
+    } else {
+        // This is a DesignSpecification:OutdoorAir:SpaceList
+        for (int dsoaCount = 1; dsoaCount <= this->numDSOA; ++dsoaCount) {
+            auto const &thisDSOA = state.dataSize->OARequirements(this->dsoaIndexes(dsoaCount));
+            int const dsoaSpaceNum = this->dsoaSpaceIndexes(dsoaCount);
+            if ((spaceNum == 0) || (spaceNum == dsoaSpaceNum)) {
+                Real64 spaceArea = state.dataHeatBal->space(this->dsoaSpaceIndexes(dsoaCount)).FloorArea;
+                sumArea += spaceArea;
+            }
+        }
+    }
+    auto &thisZone = state.dataHeatBal->Zone(zoneNum);
+    sumArea = sumArea * thisZone.Multiplier * thisZone.ListMultiplier;
+    return sumArea;
+}
+
+Real64 OARequirementsData::desFlowPerZoneArea(EnergyPlusData &state, int const zoneNum, int const spaceNum)
```

Comment:
Existing method to calculate design flow per area, easy for a simple DSOA, but needs to average across spaces for a DSAO:SpaceList.